org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/CodeActionHandlerUseWorkspaceEditTest.java

```diff
@@ -0,0 +1,253 @@
+			params.setContext(context);
+			List<Either<Command, CodeAction>> codeActions = getCodeActions(params);
+			Assert.assertNotNull(codeActions);
+		} finally {
+			cu.discardWorkingCopy();
+		}
+	}
+
+	@Test
+	@Ignore
+	public void testCodeAction_superfluousSemicolon() throws Exception{
+		ICompilationUnit unit = getWorkingCopy(
+				"src/java/Foo.java",
+				"public class Foo {\n"+
+						"	void foo() {\n"+
+						";" +
+						"	}\n"+
+				"}\n");
+
+		CodeActionParams params = new CodeActionParams();
+		params.setTextDocument(new TextDocumentIdentifier(JDTUtils.toURI(unit)));
+		final Range range = CodeActionUtil.getRange(unit, ";");
+		params.setRange(range);
+		params.setContext(new CodeActionContext(Arrays.asList(getDiagnostic(Integer.toString(IProblem.SuperfluousSemicolon), range))));
+		List<Either<Command, CodeAction>> codeActions = getCodeActions(params);
+		Assert.assertNotNull(codeActions);
+		Assert.assertEquals(1, codeActions.size());
+		Assert.assertEquals(codeActions.get(0), CodeActionKind.QuickFix);
+		Command c = getCommand(codeActions.get(0));
+		Assert.assertEquals(CodeActionHandler.COMMAND_ID_APPLY_EDIT, c.getCommand());
+		Assert.assertNotNull(c.getArguments());
+		Assert.assertTrue(c.getArguments().get(0) instanceof WorkspaceEdit);
+		WorkspaceEdit we = (WorkspaceEdit) c.getArguments().get(0);
+		List<org.eclipse.lsp4j.TextEdit> edits = we.getChanges().get(JDTUtils.toURI(unit));
+		Assert.assertEquals(1, edits.size());
+		Assert.assertEquals("", edits.get(0).getNewText());
+		Assert.assertEquals(range, edits.get(0).getRange());
+	}
+
+	@Test
+	public void test_noUnnecessaryCodeActions() throws Exception{
+		//@formatter:off
+		ICompilationUnit unit = getWorkingCopy(
+				"src/org/sample/Foo.java",
+				"package org.sample;\n"+
+				"\n"+
+				"public class Foo {\n"+
+				"	private String foo;\n"+
+				"	public String getFoo() {\n"+
+				"	  return foo;\n"+
+				"	}\n"+
+				"   \n"+
+				"	public void setFoo(String newFoo) {\n"+
+				"	  foo = newFoo;\n"+
+				"	}\n"+
+				"}\n");
+		//@formatter:on
+		CodeActionParams params = new CodeActionParams();
+		params.setTextDocument(new TextDocumentIdentifier(JDTUtils.toURI(unit)));
+		final Range range = CodeActionUtil.getRange(unit, "String foo;");
+		params.setRange(range);
+		params.setContext(new CodeActionContext(Collections.emptyList()));
+		List<Either<Command, CodeAction>> codeActions = getCodeActions(params);
+		Assert.assertNotNull(codeActions);
+		Assert.assertFalse("No need for organize imports action", containsKind(codeActions, CodeActionKind.SourceOrganizeImports));
+		Assert.assertFalse("No need for generate getter and setter action", containsKind(codeActions, JavaCodeActionKind.SOURCE_GENERATE_ACCESSORS));
+	}
+
+	private List<Either<Command, CodeAction>> getCodeActions(CodeActionParams params) {
+		return server.codeAction(params).join();
+	}
+
+	public static Command getCommand(Either<Command, CodeAction> codeAction) {
```

Comment:
duplicate with CodeActionHandlerTest.

Reply:
Fixed in da69be95d3517518867f94a1e98817b2e7f39f0f.

- https://github.com/eclipse/eclipse.jdt.ls/commit/da69be95d3517518867f94a1e98817b2e7f39f0f#diff-40c092ea2f1a4bf03e8de7891cb6625eL166
- https://github.com/eclipse/eclipse.jdt.ls/commit/da69be95d3517518867f94a1e98817b2e7f39f0f#diff-40c092ea2f1a4bf03e8de7891cb6625eL210